ams/AzureResourceProvider.cs

```diff
@@ -26,24 +29,48 @@ public AzureResourceProvider(TokenCredential credential, GlobalOptions options)
                 options.SubscriptionId,
                 options.ResourceGroup);
             _resourceGroup = armClient.GetResourceGroupResource(resourceGroupId);
+            resourceGroups = new Dictionary<string, ResourceGroupResource>();
+        }
+
+        public async Task SetResourceGroupsAsync(MediaServicesAccountResource account, CancellationToken cancellationToken)
+        {
+            var armDics = await GetResourceGroupNameAsync(account, cancellationToken);
+            if (armDics != null)
+            {
+                foreach (var arm in armDics)
+                {
+                    var clientOptions = new ArmClientOptions();
+                    clientOptions.Diagnostics.ApplicationId = $"AMSMigrate/{GetType().Assembly.GetName().Version}";
+                    var armClient = new ArmClient(_credentials, default, clientOptions);
+                    var resourceGroupId = ResourceGroupResource.CreateResourceIdentifier(
+                        _globalOptions.SubscriptionId,
```

Comment:
Instead of using the SubscriptionId that is set from the command line, it would be more flexible to figure out the storage's SubscriptionId from the Storage Id.